labels/labels.go

```diff
@@ -0,0 +1,46 @@
+package labels
+
+import (
+	"errors"
+)
+
+const (
+	// MaxLength is the maximum length we allow for labels.
+	MaxLength = 500
+
+	// Reserved is used as a prefix to separate labels that are created by
+	// loopd from those created by users.
+	Reserved = "[reserved]"
+)
+
+var (
+	// ErrLabelTooLong is returned when a label exceeds our length limit.
+	ErrLabelTooLong = errors.New("label exceeds maximum length")
+
+	// ErrReservedPrefix is returned when a label contains the prefix
+	// which is reserved for internally produced labels.
+	ErrReservedPrefix = errors.New("label contains reserved prefix")
+)
+
+// Validate checks that a label is of appropriate length and is not in our list
+// of reserved labels.
+func Validate(label string) error {
+	if len(label) > MaxLength {
+		return ErrLabelTooLong
+	}
+
+	// If the label is shorter than our reserved prefix, it cannot contain
+	// it.
+	if len(label) < len(Reserved) {
```

Comment:
can the valid label be empty?

Reply:
yes, in the case where a user doesn't provide a swap label, we can allow empty ones.